### DISCUSSION TITLE
rule to control blinds - Tasmota/wemos ESP8266

### DISCUSSION THREAD

**Discussion by jarod184** [arendst/Tasmota#15682] [Q&A] [answered]
hi all,
i'm quite a newby on tasmota, as i was using espeasy for a while.
I've recently switched to to Tasmota/mqtt/domoticz which is more reliable.
however i'm struggling with rules. i don't figure out how to do to achieve my goal.

my config: Wemos mini D1 / tasmota / 3 relays / 3 buttons
I would like to do the following behaviour to control my blind:

->when I single press button1, it power OFF relay3 and then power ON relay1 for 10 secs
->when I double press button1, it power OFF relay3 and then power ON relay1 for 20 secs

->when I single press button2, it power OFF relay3 and relay1

if someone can help me, that would be great

**Comment by barbudor** [comment, marked as answer]
Hi,
You can't changed Button_1 behavior for double-press and more. These are hard coded in Tasmota:
- Button_1 single press : toggle Relay 1 : overridable
- Button_1 double press : toggle Relay 2 : hard coded can't be changed
- Button_1 triple press : toggle Relay 3 : hard coded can't be changed
- ...

So you will have to swap your usage of Button_1 and Button_2
First detach buttons from relay by using
```
SetOption73 1
```

To catch Button_1 single press and power OFF Relay3 and 4:
```
Rule1 ON Button1#state DO backlog Power3 OFF; Power4 OFF ENDON
```

To catch Button_2 single press and power OFF relay3 and then power ON relay1 for 10 secs, or 20 Seconds for double press
```
Rule1 +ON Button2#state==10 DO backlog Power3 OFF; Power1 ON; RuleTimer1 10 ENDON
Rule1 +ON Button2#state==11 DO backlog Power3 OFF; Power1 ON; RuleTimer1 20 ENDON
Rule1 +ON Rules#Timer==1 DO Power1 OFF ENDON
```

Then enable the rule with
```
Rule1 1
```

I haven't tested but that should be ok

**Comment by jarod184** [comment]
Hi,

Thx for the help, it works like a charm.

I was missing the information for hardcoded button1

Thus, I’ve done like you said, swap button 1 and 2

 

Final config is: 

button 1 -> stop relay2 and 3

button 2 -> power OFF relay3 and then power ON relay2 for 5 secs, or 10 Seconds for double press

button 3 -> power OFF relay2 and then power ON relay3 for 5 secs, or 10 Seconds for double press

 

SetOption73 1

Rule1 ON Button1#state DO backlog Power3 OFF; Power2 OFF ENDON

 

#To catch Button_2 single press and power OFF relay3 and then power ON relay2 for 10 secs, or 20 Seconds for double press

Rule1 +ON Button2#state==10 DO backlog Power3 OFF; Power2 ON; RuleTimer1 5 ENDON

Rule1 +ON Button2#state==11 DO backlog Power3 OFF; Power2 ON; RuleTimer1 10 ENDON

Rule1 +ON Rules#Timer==1 DO Power2 OFF ENDON

 

#To catch Button_3 single press and power OFF relay2 and then power ON relay3 for 10 secs, or 20 Seconds for double press

Rule1 +ON Button3#state==10 DO backlog Power2 OFF; Power3 ON; RuleTimer1 5 ENDON

Rule1 +ON Button3#state==11 DO backlog Power2 OFF; Power3 ON; RuleTimer1 10 ENDON

Rule1 +ON Rules#Timer==1 DO Power3 OFF ENDON

 

Once again, many thx !
